siliconcompiler/core.py

```diff
@@ -461,9 +468,25 @@ def create_cmdline(self, progname, description=None, switchlist=None, input_map=
 
         parser.add_argument('-version', action='version', version=_metadata.version)
 
+        if additional_args:
+            # Add additional user specified arguments
+            for arg, arg_detail in additional_args.items():
+                parser.add_argument(arg, **arg_detail)
+
         #Grab argument from pre-process sysargs
         cmdargs = vars(parser.parse_args(scargs))
 
+        extra_params = None
+        if additional_args:
+            # Grab user specified arguments
+            extra_params = {}
+            for arg in additional_args.keys():
+                arg = arg.replace('-', '')
```

Comment:
The actual conversion algorithm is something like:
- If user supplies "dest" kwarg, use that, otherwise:
- Remove `-` from front of string
- Replace `-` with `_` in middle of string

I think just replacing `-` will result in bugs in some cases. I tried to find a nice way to grab a value (or the final translated key) from argparse based on the actual switch name, but couldn't find anything clean. It might be best to just mimic the full argparse logic, since it doesn't seem too bad (source: https://docs.python.org/3/library/argparse.html#dest)

Reply:
I'll just use the action that is returned from the add_argument call to get the dest that avoids this confusion

Reply:
Oh nice, that's the sort of solution I was looking for